Runtime/MeshOperations/CombineMeshes.cs

```diff
@@ -114,6 +114,196 @@ public static List<ProBuilderMesh> Combine(IEnumerable<ProBuilderMesh> meshes)
+                var meshFaces = mesh.facesInternal;
+                var meshSharedVertices = mesh.sharedVertices;
+                var meshSharedTextures = mesh.sharedTextures;
+                var materials = mesh.renderer.sharedMaterials;
+                var materialCount = materials.Length;
+
+                for (int i = 0; i < meshVertexCount; i++)
+                {
+                    var worldVertex = transform.TransformVertex(meshVertices[i]);
+                    vertices.Add(targetTransform.InverseTransformVertex(worldVertex));
+                }
+
+
+                foreach (var face in meshFaces)
+                {
+                    var newFace = new Face(face);
+                    newFace.ShiftIndexes(offset);
+
+                    // prevents uvs from shifting when being converted from local coords to world space
+                    if (!newFace.manualUV && !newFace.uv.useWorldSpace)
+                    {
+                        newFace.manualUV = true;
+                        autoUvFaces.Add(newFace);
+                    }
+                    var material = materials[Math.Clamp(face.submeshIndex, 0, materialCount - 1)];
+                    var submeshIndex = materialMap.IndexOf(material);
+
+                    if (submeshIndex > -1)
+                    {
+                        newFace.submeshIndex = submeshIndex;
+                    }
+                    else
+                    {
+                        if (material == null)
+                        {
+                            newFace.submeshIndex = 0;
+                        }
+                        else
+                        {
+                            newFace.submeshIndex = materialMap.Count;
+                            materialMap.Add(material);
+                        }
+                    }
+
+                    faces.Add(newFace);
+                }
+
+                foreach (var sv in meshSharedVertices)
+                {
+                    var nsv = new SharedVertex(sv);
+                    nsv.ShiftIndexes(offset);
+                    sharedVertices.Add(nsv);
+                }
+
+                foreach (var st in meshSharedTextures)
+                {
+                    var nst = new SharedVertex(st);
+                    nst.ShiftIndexes(offset);
+                    sharedTextures.Add(nst);
+                }
+
+                offset += meshVertexCount;
+            }
+
+            meshTarget.SetVertices(vertices);
+            meshTarget.faces = faces;
+            meshTarget.sharedVertices = sharedVertices;
+            meshTarget.sharedVertices = sharedTextures != null ? sharedTextures.ToArray() : null;
+            meshTarget.renderer.sharedMaterials = materialMap.ToArray();
+            meshTarget.ToMesh();
+            meshTarget.Refresh();
+
+            var returnedMesh = new List<ProBuilderMesh>() { meshTarget };
+            if (remainderMeshContributors.Count > 1)
```

Comment:
I wonder if this is necessary as I would expect the original Combine() function to handle both cases count > 1 and count == 1.

Reply:
Actually it does not.
if (!meshes.Any() || meshes.Count() < 2)
                return null;

Reply:
Oh.

@karl- , it doesn't return a single mesh if we pass a single mesh?

Reply:
No, it should return `null` if the function does not perform any action.